src/document.c

```diff
@@ -231,8 +231,14 @@ static int replyCallback(RedisModuleCtx *ctx, RedisModuleString **argv, int argc
   return REDISMODULE_OK;
 }
 
+typedef struct DocumentAddCtx {
+  RSAddDocumentCtx *aCtx;
+  RedisSearchCtx *sctx;
+} DocumentAddCtx;
```

Comment:
Please explain

Reply:
Needed to pass both arguments using a single value to `threadCallback`